### ISSUE TITLE
New "Glossary" page

### ISSUE THREAD

**Issue by Josh-Cena** [facebook/docusaurus#7670]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Description

We should have a dedicated page for all "jargons" used in the Docusaurus site, especially those UI-related. For example:

- Category link
- Category generated index page
- Docs sidebar
- Navbar
- Table of contents
- Footer
- Mobile sidebar (which is just the navbar)
- Mobile secondary menu (the one populated by the doc sidebar)
- Color mode toggle
- Much, much more

As well as some abstract technical terms like:

- Swizzling
- Plugins/themes/presets
- Theme config
- Bundler, transpiler, minifier

The UI-related terms would need some screenshots about which specific regions we are referring to. The aim is to create a unified lexicon in the community and make people's questions less ambiguous. Another gain is that some of these terms are quite hard to translate (for example, "preset" doesn't have a natural Chinese translation), so having a dedicated keyword-definition page allows people to be exposed to these unnatural terms.

This should be unversioned, so it could probably be a standalone Markdown page.

### Self-service

- [X] I'd be willing to address this documentation request myself.

**Comment by Josh-Cena** [comment]
Indeed, there are probably a few interesting questions we can put in FAQ (especially error handling and swizzling use-cases), but IMO the best way out is still putting them in the relevant guide sections where we can, since big FAQs are quite hard to navigate.

**Comment by slorber** [comment]
> Indeed, there are probably a few interesting questions we can put in FAQ (especially error handling and swizzling use-cases), but IMO the best way out is still putting them in the relevant guide sections where we can, since big FAQs are quite hard to navigate.

For me the important thing in having this FAQ is to get search indexing. Many users will simply not search for answers on GitHub, but only on our website. A long messy FAQ remains valuable for this reason, as a "buffer" before we document things better.

I don't remember which but I've seen doc sites automatically publishing some Github discussion content to their site, based on a GitHub Label (only "clean" question/answers are published)

**Comment by Josh-Cena** [comment]
Oh yeah, that would be neat.
